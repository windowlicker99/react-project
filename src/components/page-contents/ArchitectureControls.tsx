import React, { FC, useEffect, useState } from 'react';
import { Btn } from '@components/common/form-controls/Btn';
import { Badge } from '@components/common/Badge';
import { Legend } from '@components/common/Legend';
import { PageFilter } from '@components/page-contents/PageFilter';
import { CircleChart } from '@components/common/CircleChart';
import { domainFilters } from '@constants/Filters';
import { IVehicle } from '@interfaces/interfaces';
import { EBadgeTypes, ERiskLevels } from '@interfaces/enums';
import { initialVehicle } from '@constants/initialData';

interface IArchitectureControls {
  vehicle: IVehicle;
  onOpenDomainModalClick: () => void;
  onOpenHistoryView: () => void;
}

export const ArchitectureControls: FC<IArchitectureControls> = ({
  vehicle,
  onOpenDomainModalClick,
  onOpenHistoryView,
}) => {
  const [displayedVehicle, setDisplayedVehicle] = useState<IVehicle>(initialVehicle);
  const [isOpenFilters, setIsOpenFilters] = useState<boolean>(false);

  const onShowFiltersClick = () => {
    setIsOpenFilters(!isOpenFilters);
  };

  const refreshFilters = () => {
    // temporary solution. Will be removed with adding filters refresh logic
    console.log('refresh filters');
  };

  useEffect(() => {
    if (!vehicle) {
      setDisplayedVehicle(initialVehicle);
      return;
    }
    setDisplayedVehicle(vehicle);
  }, [vehicle]);

  return (
    <div className="box page-controls">
      <div className="page-controls-content">
        <div className="page-controls-item">
          <div className="mr-15">
            <Btn
              disabledBtn={!vehicle}
              btnStatus="btn-lg-primary"
              btnIcon="icon-plus"
              btnText="Create Domain"
              onClick={onOpenDomainModalClick}
            />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-lg-outline" btnText="View History" onClick={onOpenHistoryView} />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-xs-primary" btnIcon="icon-filter-blue" onClick={onShowFiltersClick} />
          </div>
          <div className="mr-15">
            <Btn btnStatus="btn-xs-secondary" btnIcon="icon-refresh" onClick={refreshFilters} />
          </div>
        </div>
        <div className="page-controls-item">
          <div className="page-controls-architecture">
            <div className="page-controls-data">
              {vehicle && (
                <div className="page-controls-info">
                  <span className="page-controls-name">{displayedVehicle.brand?.name}</span>
                  {displayedVehicle.model}
                </div>
              )}
            </div>
            <div className="page-controls-data">
              Risk Level
              <Badge
                badgeType={EBadgeTypes[displayedVehicle.riskLevel]}
                badgeText={ERiskLevels[displayedVehicle.riskLevel]}
              />
            </div>
            <div className="page-controls-data">
              Progress
              <div className="chart">
                <CircleChart percentage={displayedVehicle.raProgress} />
              </div>
            </div>
            <div className="page-controls-data">
              <Legend isDisabled={!vehicle} legendType="Eth" legendName="Ethernet" />
              <Legend isDisabled={!vehicle} legendType="CAN" legendName="CAN" />
              <Legend isDisabled={!vehicle} legendType="LIN" legendName="LIN" />
            </div>
          </div>
        </div>
      </div>
      <PageFilter filters={domainFilters} isOpen={isOpenFilters} close={onShowFiltersClick} />
    </div>
  );
};
