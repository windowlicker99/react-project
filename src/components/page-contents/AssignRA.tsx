import React, { FC, useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { useFormik } from 'formik';
import { ModalCenter } from '@components/modules/ModalCenter';
import { AssignTicketModalContent } from '@components/modal-content/AssignTicketModalContent';
import { selectCurrentVehicle } from '@store/selectors/vehicle';
import { getUsers } from '@utils/getUsers';
import { convertUsersToOptions } from '@utils/converters';
import { getElement } from '@utils/getElement';
import { createTicket } from '@utils/createTicket';
import { ICurrentElement, IOption } from '@interfaces/interfaces';
import { TElement } from '@interfaces/types';
import { EAsideActualElement, ETicketsTypes } from '@interfaces/enums';

interface IAssignRA {
  element: ICurrentElement;
  isOpen: boolean;
  onClose: () => void;
}

export const AssignRA: FC<IAssignRA> = ({ element, isOpen, onClose }) => {
  const defaultOption = { label: 'Not Assigned', value: 'NaN' };
  const [currentElement, setCurrentElement] = useState<TElement>(null);
  const [usersOptions, setUsersOptions] = useState<IOption[]>([defaultOption]);

  const vehicleId = useSelector(selectCurrentVehicle);

  const getUsersOptions = async () => {
    const users = await getUsers();
    const options = convertUsersToOptions(users);
    setUsersOptions([defaultOption, ...options]);
  };

  const getCurrentElement = async () => {
    const newElement = await getElement(element);
    setCurrentElement(newElement);
  };

  useEffect(() => {
    if (element) {
      getCurrentElement();
    }
  }, [element]);

  useEffect(() => {
    getUsersOptions();
  }, []);

  const formik = useFormik({
    initialValues: { user: defaultOption, message: '' },
    onSubmit: async ({ user: { value: userId }, message }, { resetForm }) => {
      try {
        await createTicket({
          name: `Risk Assessment for ${currentElement?.name}`,
          type: ETicketsTypes.riskAssessment,
          elementName: currentElement.name,
          elementType: EAsideActualElement[element.type],
          elementId: element.id,
          userId,
          vehicleId,
          message,
        });
      } catch (error) {
        console.log(error);
      } finally {
        resetForm();
        onClose();
      }
    },
  });

  const handleSelectChange = (name: string, value: IOption) => formik.setFieldValue(name, value);

  const onApplyRAModal = () => {
    formik.submitForm();
  };

  const onCloseRAModal = () => {
    formik.resetForm();
    onClose();
  };

  return (
    <ModalCenter
      title="Assign Ticket"
      subtitle={`Risk Assessment for ${currentElement?.name}`}
      isOpen={isOpen}
      disabledApplyBtn={!formik.values.user || formik.values.user.value === defaultOption.value}
      onApply={onApplyRAModal}
      onClose={onCloseRAModal}
    >
      <AssignTicketModalContent
        values={formik.values}
        options={usersOptions}
        onChange={formik.handleChange}
        onSelectChange={handleSelectChange}
      />
    </ModalCenter>
  );
};
