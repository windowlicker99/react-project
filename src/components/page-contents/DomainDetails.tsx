import React, { useState, FC } from 'react';
import { useFormik } from 'formik';
import { useSelector } from 'react-redux';
import { ModalAside } from '@components/modules/ModalAside';
import { DomainDetailsContent } from '@components/modal-content/DomainDetailsContent';
import { ModalFooter } from '@components/modal-content/modal-aside/ModalFooter';
import { ModalHeader } from '@components/modal-content/modal-aside/ModalHeader';
import { selectUser } from '@store/selectors/auth';
import { getFieldsValues } from '@utils/getFieldsValues';
import { domainValidationSchema } from '@utils/validations';
import { addNewDomain } from '@utils/addNewDomain';
import { domainCreationElements } from '@constants/formsElements';
import { IOption, IVehicle } from '@interfaces/interfaces';
import { EElementCreationTabs } from '@/interfaces/enums';
import { DomainChooseContent } from '../modal-content/DomainChooseCollection';
import { elementDetailsTabs } from '@/constants/tabsLists';
import { ResultContent } from '../modal-content/ResultContent';
import { backgroundColors } from '@/constants/color';

interface IDomainDetails {
  vehicle: IVehicle;
  isOpenDomainModal: boolean;
  onCloseDomainModalClick: () => void;
}

export const DomainDetails: FC<IDomainDetails> = ({ vehicle, isOpenDomainModal, onCloseDomainModalClick }) => {
  const initialValues = getFieldsValues(Object.values(domainCreationElements));
  const [successfullyAdded, setSuccessfullyAdded] = useState<string>('');
  const [detailsForm, setDomainDetailsForm] = useState<string>(EElementCreationTabs.create);

  const user = useSelector(selectUser);

  const formik = useFormik({
    initialValues,
    validationSchema: domainValidationSchema,
    onSubmit: async ({ name, domainUsage, connectivity, color, communication: selectedCommunication }) => {
      try {
        const communication: string[] = selectedCommunication.map(({ label }: IOption) => label);
        const onlineConnectivity = connectivity ? 'Online' : 'Offline';
        await addNewDomain({
          name,
          domainUsage,
          onlineConnectivity,
          communication,
          color,
          background: backgroundColors[color as keyof typeof backgroundColors],
          authorId: user.id,
          vehicleId: vehicle.id,
        });
        setSuccessfullyAdded(
          `${name} was successfully added to ${vehicle.brand?.name} | ${vehicle.model} | ${vehicle.platform} architecture scheme!`
        );
      } catch (error) {
        console.log(error);
        onCloseDomainModalClick();
        setSuccessfullyAdded('');
      }
    },
  });

  const handleSelectChange = (name: string, value: IOption) => formik.setFieldValue(name, value);

  const onApplyModal = () => {
    formik.validateForm();
    formik.submitForm();
  };

  const onCloseModal = () => {
    formik.resetForm();
    onCloseDomainModalClick();
  };

  const onConfirmClick = () => {
    setSuccessfullyAdded('');
    onCloseModal();
  };

  return successfullyAdded ? (
    <ModalAside isOpen={isOpenDomainModal}>
      <ResultContent title="Success" content={successfullyAdded} onConfirm={onConfirmClick} />
    </ModalAside>
  ) : (
    <ModalAside
      isOpen={isOpenDomainModal}
      header={
        <ModalHeader
          title="Domain Details"
          activeTab={detailsForm}
          tabsList={elementDetailsTabs}
          onClose={onCloseModal}
          onChooseTabClick={setDomainDetailsForm}
        />
      }
      footer={
        <ModalFooter
          applyBtn="Create"
          onApply={onApplyModal}
          showDuplicateSwitch={detailsForm === EElementCreationTabs.choose}
        />
      }
    >
      {detailsForm === EElementCreationTabs.create ? (
        <DomainDetailsContent {...formik} handleSelectChange={handleSelectChange} />
      ) : (
        <DomainChooseContent />
      )}
    </ModalAside>
  );
};
