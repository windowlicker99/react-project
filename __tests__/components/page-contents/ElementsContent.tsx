import React from 'react';
import * as reactRedux from 'react-redux';
import { render } from '@testing-library/react';
import { ElementsContent } from '@components/page-contents/ElementsContent';

describe('ElementsContent tests', () => {
  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
  });

  it('Should render', () => {
    useSelectorMock.mockReturnValue([
      { label: 'Name', name: 'name' },
      { label: 'Date', name: 'date' },
    ]);
    useDispatchMock.mockReturnValue(jest.fn());

    const { container } = render(<ElementsContent />);
    expect(container.getElementsByClassName('elements-content').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    useSelectorMock.mockReturnValue([
      { label: 'Name', name: 'name' },
      { label: 'Date', name: 'date' },
    ]);
    useDispatchMock.mockReturnValue(jest.fn());

    const { asFragment } = render(<ElementsContent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
