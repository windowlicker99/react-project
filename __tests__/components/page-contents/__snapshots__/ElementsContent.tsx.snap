// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementsContent tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="box-table elements-content"
  >
    <table
      class="table"
    >
      <thead
        class="table-header"
      >
        <tr
          class="table-row"
        >
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Model
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Platform
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Name
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Domain Usage
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Communication
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Online Connectivity
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Author
              </span>
              <div
                class="table-filter"
              >
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-up"
                  />
                </button>
                <button
                  class="table-filter-btn"
                  type="button"
                >
                  <span
                    class="icon-arrow-full-down"
                  />
                </button>
              </div>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Last Update
              </span>
            </div>
          </th>
          <th
            class="table-col"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Color
              </span>
            </div>
          </th>
          <th
            class="table-col sticky"
          >
            <div
              class="table-header-inner"
            >
              <span
                class="table-header-title"
              >
                Actions
              </span>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="table-body"
      >
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            20.3
          </td>
          <td
            class="table-col"
          >
            30.5
          </td>
          <td
            class="table-col"
          >
            23.5
          </td>
          <td
            class="table-col"
          >
            40.3
          </td>
          <td
            class="table-col"
          >
            23.5
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round green"
              style="background-color: green;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            50.2
          </td>
          <td
            class="table-col"
          >
            40.63
          </td>
          <td
            class="table-col"
          >
            45.23
          </td>
          <td
            class="table-col"
          >
            39.3
          </td>
          <td
            class="table-col"
          >
            40.3
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round red"
              style="background-color: red;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round yellow"
              style="background-color: yellow;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
        <tr
          class="table-row"
        >
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            25.4
          </td>
          <td
            class="table-col"
          >
            30.2
          </td>
          <td
            class="table-col"
          >
            33.3
          </td>
          <td
            class="table-col"
          >
            36.7
          </td>
          <td
            class="table-col"
          >
            2021 TSI
          </td>
          <td
            class="table-col"
          >
            <div
              class="avatar-card"
            >
              <div
                class="avatar"
              >
                <img
                  alt="avatar"
                  class="avatar-img"
                  src=""
                />
              </div>
              <span
                class="avatar-name"
              >
                John Stone
              </span>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="date"
            >
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-calendar"
                  />
                </div>
                <span
                  class="date-data"
                >
                  01/01/2021
                </span>
              </div>
              <div
                class="date-item"
              >
                <div
                  class="date-icon"
                >
                  <span
                    class="icon-time"
                  />
                </div>
                <span
                  class="date-data"
                >
                  12:00 AM
                </span>
              </div>
            </div>
          </td>
          <td
            class="table-col"
          >
            <div
              class="badge-round azure"
              style="background-color: azure;"
            />
          </td>
          <th
            class="table-col sticky"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-xs-primary mr-15"
                type="button"
              >
                <span
                  class="btn-icon icon-architecture-blue"
                />
                <span
                  class="btn-text"
                />
              </button>
              <button
                class="btn btn-xs-primary mr-40"
                type="button"
              >
                <span
                  class="btn-icon icon-elements"
                />
                <span
                  class="btn-text"
                />
              </button>
            </div>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
